### PR TITLE
feat: match latest vm spec PR

### PR DESCRIPTION
**Pull request by icorderi**
### Motivation

Spec changed, we change. 

### Proposed changes

- using implementer discretion on:
    * Send signature
    * Grouping of exposed methods

### CODE REVIEW

File: internal/pkg/vm/actor/builtin/initactor/init.go [hunk 1]
```diff
@@ -270,7 +270,7 @@ func (a *Impl) Exec(rt invocationContext, codeCID cid.Cid, params []interface{})
 	}
 
 	// send message containing actor's initial balance to construct it with the given params
-	_, _, err = rt.Runtime().Send(idAddr, types.ConstructorMethodID, rt.Message().Value, params)
+	_, _, err = rt.Runtime().Send(idAddr, types.ConstructorMethodID, rt.OriginalMessage().Value, params)
```
Comment (anorth) [inline internal/pkg/vm/actor/builtin/initactor/init.go]: You can change this to rt.ValueTransferred() and remove the message.

File: internal/pkg/vm/actor/builtin/initactor/init.go [hunk 2]
```diff
@@ -126,7 +126,7 @@ type invocationContext interface {
 	runtime.InvocationContext
 	AddressForNewActor() (address.Address, error)
 	CreateNewActor(addr address.Address, code cid.Cid) error
-	Message() *types.UnsignedMessage
+	OriginalMessage() *types.UnsignedMessage
```
Comment (anorth) [inline internal/pkg/vm/actor/builtin/initactor/init.go]: As discussed, let's say `LegacyMessage`. This is the immediate message, not the top-level one.

File: internal/pkg/vm/actor/builtin/miner/miner.go [hunk 3]
```diff
@@ -470,7 +470,7 @@ func (*Impl) AddAsk(ctx invocationContext, price types.AttoFIL, expiry *big.Int)
 
 	var state State
 	out, err := actor.WithState(ctx, &state, func() (interface{}, error) {
-		if ctx.Message().From != state.Worker {
+		if ctx.OriginalMessage().From != state.Worker {
```
Comment (anorth) [inline internal/pkg/vm/actor/builtin/miner/miner.go]: You can change this to ctx.Caller and remove the message. Same goes for all the other uses of From in this file (and everywhere else).

File: internal/pkg/vm/actor/builtin/miner/miner.go [hunk 4]
```diff
@@ -730,7 +730,7 @@ func (a *Impl) CommitSector(ctx invocationContext, sectorID uint64, commD, commR
 			copy(req.CommR[:], commR)
 			copy(req.CommRStar[:], commRStar)
 			req.Proof = proof
-			req.ProverID = sectorbuilder.AddressToProverID(ctx.Message().To)
+			req.ProverID = sectorbuilder.AddressToProverID(ctx.OriginalMessage().To)
```
Comment (anorth) [inline internal/pkg/vm/actor/builtin/miner/miner.go]: FYI I'm adding `Receiver` to the spec, which can replace this use of message

Comment (icorderi) [reply internal/pkg/vm/actor/builtin/miner/miner.go]: 👍 thanks!

**Pull request merged**